File: src/main/java/net/vidageek/games/regex/task/ValidationIfAllCapture.java
package net.vidageek.games.regex.task;

import static net.vidageek.games.regex.task.MatcherTargets.from;

import java.util.regex.Matcher;

import net.vidageek.games.task.JudgedTask;
import net.vidageek.games.task.status.Failed;
import net.vidageek.games.task.status.Ok;

public class ValidationIfAllCapture implements GroupValidation {

	private final String[] captureGroupTargets;

	public ValidationIfAllCapture(final String... captureGroupTargets) {
		this.captureGroupTargets = captureGroupTargets;
	}

	public JudgedTask judge(final String challenge, final Matcher matcher) {
		if (matcher.groupCount() != captureGroupTargets.length) {
			return new Failed("A regex " + from(challenge).asHtml() + " n&atilde;o cont&eacute;m "
					+ captureGroupTargets.length + " grupo(s) de captura.");
		}
		return new Ok();
	}

}

File: src/main/scala/vggames/regex/task/ValidationIfAllCapture.scala
package vggames.regex.task;

import java.util.regex.Matcher

import vggames.regex.task.MatcherTargets.from
import vggames.shared.task.status.{Ok, Failed}
import vggames.shared.task.JudgedTask

class ValidationIfAllCapture(captureGroupTargets : String*) extends GroupValidation {

  def judge(challenge : String, matcher : Matcher) : JudgedTask = {
    if (matcher.groupCount() != captureGroupTargets.length) {
      return new Failed("A regex " + from(challenge).asHtml() + " n&atilde;o cont&eacute;m "
        + captureGroupTargets.length + " grupo(s) de captura.");
    }
    return new Ok();
  }

}
